File: UserNotifications/UserNotificationsObjCSample/UserNotificationsObjCSample/MainViewController.m
//
//  MainViewController.m
//  UserNotificationsObjCSample
//
//  Created by Nikolay Andonov on 11/9/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

#import "MainViewController.h"
#import "UserNotificationsHandler.h"

@interface MainViewController ()
@property (weak, nonatomic) IBOutlet UIDatePicker *datePicker;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)scheduleButtonAction:(id)sender {
    
    NSDate *selectedDate = self.datePicker.date;
    NSLog(@"Selected date: %@",selectedDate);
    [[UserNotificationsHandler sharedHandler] scheduleNotificationAtDate:selectedDate];
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:nil message:@"The reminder was scheduled!" preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}


@end

File: UserNotifications/UserNotificationsSwiftSample/UserNotificationsSwiftSample/MainViewController.swift
//
//  MainViewController.swift
//  UserNotificationsSwiftSample
//
//  Created by Nikolay Andonov on 11/9/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

import UIKit

class MainViewController: UIViewController {

    @IBOutlet weak var datePicker: UIDatePicker!
    
    override func viewDidLoad() {
        super.viewDidLoad()

    }
    
    @IBAction func scheduleButtonAction(_ sender: Any) {
        
        let selectedDate = datePicker.date
        print("Selected date: \(selectedDate)")
        UserNotificationsHandler.sharedHandler.scheduleNotification(at: selectedDate)
        
        let alert = UIAlertController(title: nil, message: "The reminder was schedule!", preferredStyle: UIAlertControllerStyle.alert)
        alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
        self.present(alert, animated: true, completion: nil)
    }
}
